### PR TITLE
Trade project

### PR DESCRIPTION
**Pull request by Mahmoudmohamedalbanhawi**

### CODE REVIEW

File: trade/assets/js/index.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    const currency2Dropdown = document.getElementById('currency2');
+    const currencyNamesDiv = document.getElementById('currencyNames');
+    const ctx = document.getElementById('currencyChart').getContext('2d');
+    const timeControls = document.getElementById('timeControls');
+    const avgexchangerate = document.getElementById("price");
+    const percentage = document.getElementById("percentage");
+    const loadingIndicator = document.getElementById('loadingIndicator');
+    const flag1 = document.querySelector(".flag1");
+    const flag2 = document.querySelector(".flag2");
+    const storedCurrency1 = localStorage.getItem('selectedCurrency1');
```
Comment (MostafaAshrafMohamedKamal) [inline trade/assets/js/index.js:16]: all localStorage that you use not important and not help in minimize api calls

File: trade/assets/js/index.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    const storedLastPrice = localStorage.getItem('lastPrice');
+    const storedPercentageChange = localStorage.getItem('percentageChange');
+    const storedInterval = localStorage.getItem('selectedInterval') || '15m';
+    let currentTimeRange = storedInterval;
+
+    if (storedCurrency1) currency1Dropdown.value = storedCurrency1;
+    if (storedCurrency2) currency2Dropdown.value = storedCurrency2;
+
+    if (storedLastPrice && storedPercentageChange) {
+        percentage.innerHTML = `<span class="${storedPercentageChange < 0 ? 'text-danger' : 'text-success'}">${storedLastPrice} </span> <span class="${storedPercentageChange < 0 ? 'text-danger' : 'text-success'}">(${parseFloat(storedPercentageChange).toFixed(2)}%)</span>`;
```
Comment (MostafaAshrafMohamedKamal) [inline trade/assets/js/index.js]: use classList and textContent instead replace all html

File: trade/assets/js/index.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            hideLoadingIndicator(loadingIndicator);
+            displayDropdownList(currency1Dropdown, currencies, storedCurrency1 || 'EUR');
+            displayDropdownList(currency2Dropdown, currencies, storedCurrency2 || 'USD');
+            setFlag(currency1Dropdown.value, flag1);
+            setFlag(currency2Dropdown.value, flag2);
+            updateChart(currentTimeRange);
+        })
+        .catch(error => {
+            hideLoadingIndicator(loadingIndicator);
+            console.error('Error fetching the data:', error);
```
Comment (MostafaAshrafMohamedKamal) [inline trade/assets/js/index.js:42]: We must show the user what happened, so display to the user the error (user-friendly message)

File: trade/assets/js/service.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+
+
+export function fetchCurrencies() {
```
Comment (MostafaAshrafMohamedKamal) [inline trade/assets/js/service.js]: use const var for baseUrl = "https://marketdata.tradermade.com/api/v1/" and apiKay = "i4ydISb-yX--K-I-AMjV"

File: trade/assets/js/utils.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+   };
+export function showLoadingIndicator(loadingIndicator) {
+    loadingIndicator.style.display = 'block';
+}
+
+export function hideLoadingIndicator(loadingIndicator) {
+    loadingIndicator.style.display = 'none';
+}
+
+export function setFlag(flag, flagElement) {
```
Comment (MostafaAshrafMohamedKamal) [inline trade/assets/js/utils.js]: must handle if the flag not found

File: trade/assets/js/index.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
+// index.js
+
+import { fetchCurrencies, fetchTimeSeries } from './service.js';
```
Comment (MostafaAshrafMohamedKamal) [inline trade/assets/js/index.js]: dont use index.js as module direct add it in html and remove all export that you add in all files

File: task/index.html [hunk 7]
```diff
@@ -1,11 +0,0 @@
-<!DOCTYPE html>
-<html lang="en">
-<head>
-    <meta charset="UTF-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>js 1st task</title>
-</head>
-<body>
-    <script src="main.js"></script>
```
Comment (MostafaAshrafMohamedKamal) [inline task/index.html:9]: Directly add it in html (order is important).
<script src="https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/chart.js"></script>

    <script src="./assets/js/utils.js"></script>
    <script src="./assets/js/service.js"></script>
    <script src="./assets/js/index.js"></script>

File: trade/index.html [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+   
+
+   <div id="loadingIndicator" >
+    <div class="spinner-border" role="status">
+        <span class="visually-hidden">Loading...</span>
+    </div>
+</div>
+
+   
+    <script src="https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js"></script>
```
Comment (MostafaAshrafMohamedKamal) [inline trade/index.html:72]: Directly add it in html (order is important).
<script src="https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/chart.js"></script>

    <script src="./assets/js/utils.js"></script>
    <script src="./assets/js/service.js"></script>
    <script src="./assets/js/index.js"></script>

File: trade/assets/js/index.js (hunk 1 above)
Comment (Mahmoudmohamedalbanhawi) [reply trade/assets/js/index.js:16]: done

File: trade/assets/js/index.js (hunk 2 above)
Comment (Mahmoudmohamedalbanhawi) [reply trade/assets/js/index.js]: done

File: trade/assets/js/index.js (hunk 3 above)
Comment (Mahmoudmohamedalbanhawi) [reply trade/assets/js/index.js:42]: done

File: trade/assets/js/service.js (hunk 4 above)
Comment (Mahmoudmohamedalbanhawi) [reply trade/assets/js/service.js]: done

File: trade/assets/js/utils.js (hunk 5 above)
Comment (Mahmoudmohamedalbanhawi) [reply trade/assets/js/utils.js]: done

File: trade/assets/js/index.js (hunk 6 above)
Comment (Mahmoudmohamedalbanhawi) [reply trade/assets/js/index.js]: done

File: trade/index.html (hunk 8 above)
Comment (Mahmoudmohamedalbanhawi) [reply trade/index.html:72]: done

**Pull request merged**